quickbase.py

```diff
@@ -210,34 +214,37 @@ def request(self, action, database, request, required=None, ticket=True,
         request['encoding'] = 'UTF-8'
         request['msInUTC'] = 1
         if self.realmhost:
-            request['realmhost'] = self.realmhost
+            request['realmhost'] = self.realmhost 
         data = self._build_request(**request)
         headers = {
             'Content-Type': 'application/xml',
             'QUICKBASE-ACTION': 'API_' + action,
-        }
-        request = requests.post(url, data, headers=headers)
-        response = request.content
-        encoding = chardet.detect(response)['encoding']
-
-        if encoding != 'utf-8':
-            response = response.decode(encoding, 'replace').encode('utf-8')
-
-        try:
-            parsed = etree.fromstring(response)
-        except etree.XMLSyntaxError as e:
-            raise XMLError(-1, e, response=response)
-        except etree.DocumentInvalid as e:
-            raise XMLError(-1, e, response=response)
-
-        error_code = parsed.findtext('errcode')
-        if error_code is None:
-            raise ResponseError(-4, '"errcode" not in response', response=response)
-        if error_code != '0':
-            error_text = parsed.find('errtext')
-            error_text = error_text.text if error_text is not None else '[no error text]'
-            raise ResponseError(error_code, error_text, response=response)
-
+        }   
+        request = requests.post(url, data, headers=headers, stream=True)
+        if request.status_code == 200:
+            with open(".request", 'wb') as f:
+                for chunk in request.iter_content(2048):
+                    encode = chardet.detect(chunk)['encoding']
```

Comment:
It's been a while since I have touched any Quickbase stuff, but you may just want to drop chardet altogether and force unicode strings.

```
if not isinstance(response, unicode):
    response = response.decode('utf-8', errors='ignore')
```
Note that now your string going into `etree.fromstring` is a unicode object and not a bytestring, and may not contain an XML encoding declaration as explained here http://lxml.de/parsing.html#python-unicode-strings